From 70d39dab2e10b54a46992d38ed999d1bcf0b41ff Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:27 -0700
Subject: [PATCH 1652/1852] v4l: xilinx: remapper: Fix warnings for
 CHECKED_RETURN

commit 8d1765260ac27402a3ab9a26ddb0868575cf2aa7 from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the CHECKED_RETURN type.

The actual warnings are:
For xremap_probe():
Event check_return: 	Calling "clk_prepare_enable" without checking
return value (as is done elsewhere 129 out of 152 times).

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-remapper.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-remapper.c b/drivers/media/platform/xilinx/xilinx-remapper.c
index 66ce55204414..b1db86181a8b 100644
--- a/drivers/media/platform/xilinx/xilinx-remapper.c
+++ b/drivers/media/platform/xilinx/xilinx-remapper.c
@@ -489,7 +489,11 @@ static int xremap_probe(struct platform_device *pdev)
 	if (IS_ERR(xremap->xvip.clk))
 		return PTR_ERR(xremap->xvip.clk);
 
-	clk_prepare_enable(xremap->xvip.clk);
+	ret = clk_prepare_enable(xremap->xvip.clk);
+	if (ret) {
+		dev_err(&pdev->dev, "failed to enable clk (%d)\n", ret);
+		return ret;
+	}
 
 	/* Initialize V4L2 subdevice and media entity */
 	subdev = &xremap->xvip.subdev;
-- 
2.31.1

